### PR TITLE
feat: implement PatchTST v2 architecture

### LINKED ISSUE #2067: [ENH] Add v2 interface support for PatchTST
Referencing #1992

This issue tracks adding v2 interface support for the PatchTST model from the thuml Time-Series-Library.

The model currently does not have a v2 package. The plan is to implement the PatchTST architecture following the v2 pattern used by DLinear, SAMformer, and TFT.

CC: @fkiraly @PranavBhatP @phoeenniixx 

I would like to work on this, please assign it to me.

### PR DESCRIPTION
**Pull request by amruth6002**
#### Reference Issues/PRs
Closes #2067

#### What does this implement/fix? Explain your changes.
This PR introduces the **PatchTST** architecture to the `pytorch-forecasting` v2 interface.

The implementation adapts the core "Channel Independence" patching mechanism and self-attention encoder from the original `thuml` implementation to align with the `TslibBaseModel` standards.

**Key Additions:**
1. **Layers**: Added [PatchEmbedding](cci:2://file:///home/amruth/amruth/projects/pytorch-forecasting/pytorch_forecasting/layers/_embeddings/_patch_embedding.py:16:0-113:38), [SelfAttnEncoder](cci:2://file:///home/amruth/amruth/projects/pytorch-forecasting/pytorch_forecasting/layers/_encoders/_self_attn_encoder.py:11:0-62:23), and [SelfAttnEncoderLayer](cci:2://file:///home/amruth/amruth/projects/pytorch-forecasting/pytorch_forecasting/layers/_encoders/_self_attn_encoder_layer.py:11:0-86:38) to support the architecture without cross-attention components.
2. **Core Model**: Implemented [PatchTST](cci:2://file:///home/amruth/amruth/projects/pytorch-forecasting/pytorch_forecasting/models/patchtst/_patchtst_v2.py:93:0-404:41) inside [models/patchtst/_patchtst_v2.py](cci:7://file:///home/amruth/amruth/projects/pytorch-forecasting/pytorch_forecasting/models/patchtst/_patchtst_v2.py:0:0-0:0). It correctly handles instance normalization, permuting/patching the sequences, and supports Quantile Loss resizing configurations.
3. **Package Wrapper**: Created [PatchTST_pkg_v2](cci:2://file:///home/amruth/amruth/projects/pytorch-forecasting/pytorch_forecasting/models/patchtst/_patchtst_pkg_v2.py:7:0-149:21) for `sktime` compatibility and automated testing scaffolding.
4. **Tests**: Added [test_patchtst_v2.py](cci:7://file:///home/amruth/amruth/projects/pytorch-forecasting/tests/test_models/test_patchtst_v2.py:0:0-0:0) verifying point forecasting, univariate flows, and quantile loss architectures via standard parameterized `pytest` conventions.

All pre-commit hooks and unit tests pass successfully.

### CODE REVIEW

File: pytorch_forecasting/models/patchtst/_patchtst_v2.py [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+from pytorch_forecasting.layers._embeddings._patch_embedding import PatchEmbedding
+from pytorch_forecasting.layers._encoders._self_attn_encoder import SelfAttnEncoder
+from pytorch_forecasting.layers._encoders._self_attn_encoder_layer import (
+    SelfAttnEncoderLayer,
+)
+from pytorch_forecasting.metrics import QuantileLoss
+from pytorch_forecasting.models.base._tslib_base_model_v2 import TslibBaseModel
+
+
+class _Transpose(nn.Module):
```
Comment (PranavBhatP) [inline pytorch_forecasting/models/patchtst/_patchtst_v2.py]: Please move this to the `_layers` folder under the correct subdirectory, if it already exists, reuse.

File: pytorch_forecasting/models/patchtst/_patchtst_v2.py [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+        self.dims = dims
+        self.contiguous = contiguous
+
+    def forward(self, x):
+        if self.contiguous:
+            return x.transpose(*self.dims).contiguous()
+        return x.transpose(*self.dims)
+
+
+class _FlattenHead(nn.Module):
```
Comment (PranavBhatP) [inline pytorch_forecasting/models/patchtst/_patchtst_v2.py]: Same comment here.

File: pytorch_forecasting/models/patchtst/_patchtst_v2.py (hunk 1 above)
Comment (amruth6002) [reply pytorch_forecasting/models/patchtst/_patchtst_v2.py]: done , please recheck.

File: pytorch_forecasting/models/patchtst/_patchtst_v2.py (hunk 2 above)
Comment (amruth6002) [reply pytorch_forecasting/models/patchtst/_patchtst_v2.py]: done , please recheck

File: pytorch_forecasting/models/patchtst/_patchtst_pkg_v2.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+    @classmethod
+    def get_datamodule_cls(cls):
+        """Get the underlying DataModule class."""
+        from pytorch_forecasting.data._tslib_data_module import TslibDataModule
+
+        return TslibDataModule
+
+    @classmethod
+    def _get_test_datamodule_from(cls, trainer_kwargs):
```
Comment (phoeenniixx) [inline pytorch_forecasting/models/patchtst/_patchtst_pkg_v2.py]: I dont think we need this method?

Comment (amruth6002) [reply pytorch_forecasting/models/patchtst/_patchtst_pkg_v2.py]: That method was a leftover from an earlier testing pattern and is not used by the current v2 automated test suite. I've removed it in the latest commit.
